google/cloud/storage/internal/metadata_parser.cc

```diff
@@ -22,6 +22,29 @@ namespace cloud {
 namespace storage {
 inline namespace STORAGE_CLIENT_NS {
 namespace internal {
+bool ParseBoolField(nl::json const& json, char const* field_name) {
```

Comment:
I think it's probably worth adding docstrings to these `Parse<Foo>Field` methods mentioning that if the field at `field_name` isn't found, a default value of insert-value-here is returned.

I'd also note that the above behavior is sort of surprising.  I'd expect this to either throw an error or return an empty Optional if the field wasn't present.  Thoughts?

Reply:
Done.